File: scripts/mergeguard_demo/webhook.py
"""Billing webhook handler — no accompanying test file in this PR."""

from .billing import charge_customer, refund_charge


def handle_billing_event(event: dict) -> dict:
    event_type = event.get("type")
    if event_type == "payment_intent.requested":
        return charge_customer(
            amount=event["amount"],
            currency=event.get("currency", "usd"),
            customer_id=event["customer_id"],
        )
    if event_type == "charge.refund":
        return refund_charge(event["payment_intent_id"])
    return {"status": "ignored"}

File: src/lib/mergeguard-demo/webhook.ts
/**
 * /webhooks/billing endpoint. Handles incoming Stripe webhook events.
 * No accompanying test file in this PR — Stage 5 evidence mapper
 * should flag this.
 */
import { chargeCustomer, refundCharge } from "./billing.js";
import type { Request, Response } from "express";

export async function handleBillingWebhook(req: Request, res: Response): Promise<void> {
  const event = req.body;
  switch (event.type) {
    case "payment_intent.succeeded":
      await chargeCustomer(event.data, 3);
      break;
    case "charge.refunded":
      await refundCharge(event.data.payment_intent);
      break;
    default:
      res.status(200).send("ignored");
      return;
  }
  res.status(200).send("ok");
}
